src/confluent_kafka/schema_registry/json_schema.py

```
@@ -43,6 +43,36 @@ def __exit__(self, *args):
         return False
 
 
+def _id_of(schema):
+    """
+    Returns the schema id if present otherwise None.
+    :param schema: Schema to return id of.
+    :return: Id of schema if present otherwise None.
+    """
+    return schema.get('$id', "None")
```

shouldn't be `None` here, without quotes?

---

Thx, I've updated this.

---

Makes sense. Updated.

---

If "id" field is mandatory, shouldn't we raise the error?

---

It's not a mandatory field, but is used as a best practice and is needed if you need to reference other schemas.

---

https://json-schema.org/understanding-json-schema/structuring.html#id

---

Probably for our use case it's mandatory, we're using it as the key in a map, it could throw a ValueError in that case

---

Yeah, so should we throw an exception here?

---

Thanks. I've updated the code to use ref.name as the id.